### PR TITLE
fix(core,cli): frame-aware delegation frontier and narrowed bare collect

### LINKED ISSUE #397: Narrow idempotent bare collect behavior
## Problem

`packages/cli/src/commands/collect.ts` treats any bare `rd collect` on a non-DELEGATE cursor as `already-aggregated` and exits successfully.

That covers the intended post-auto-aggregation case, but it also masks misuse on ordinary non-DELEGATE active steps that were never delegated aggregation successors.

## Scope

- Distinguish true post-auto-aggregation state from ordinary non-DELEGATE active steps.
- Preserve idempotent behavior after delegated aggregation advances.
- Add a negative test for bare `rd collect` on an ordinary non-DELEGATE step.

## Acceptance Criteria

- Bare `rd collect` after auto-aggregation remains idempotent.
- Bare `rd collect` on unrelated non-DELEGATE steps returns an error such as `NOT_DELEGATE_STEP`.
- Tests cover both positive and negative bare-collect paths.

### LINKED ISSUE #400: Make bare delegate token echo frame-aware and core-owned
## Problem

`packages/cli/src/commands/delegate.ts` reconstructs delegation frontier entries without preserving frame scope. It includes every non-cancelled delegation and keys entries only as `${state.step}.${substep.id}`, dropping `substep.frameKey`.

In a `FOR` step or repeated-frame case, bare `rd delegate` can echo a token from another frame while the resolver is targeting the current active frame.

This is also an architecture smell: frame-aware delegation targeting should be core-owned or exposed through a core helper, not reconstructed inside the CLI.

## Scope

- Move or expose frame-aware frontier derivation in `@rundown-org/core`.
- Update CLI to consume the core helper instead of reconstructing frame-sensitive data.
- Add tests with multiple `substepStates` for the same substep id across different frames.

## Acceptance Criteria

- Bare `rd delegate` echoes the token for the active frame only.
- A repeated-frame or `FOR` case cannot return a token from another frame.
- CLI remains a thin wrapper and does not own frame-sensitive delegation logic.

### LINKED ISSUE #402: Expand core edge-case coverage for delegation and claim tombstones
## Problem

Core tests cover the main delegation/session paths but miss important edge branches introduced or affected by this branch.

Missing `resolveDelegateTarget` coverage includes:

- active delegation without a frontier token
- mixed active/issuable precedence
- done substeps
- active-frame isolation
- RD-814 for `DELEGATE` without a runbook

Missing `SessionService` coverage includes:

- stopped-child terminal tombstone behavior
- `pruneClaimsForChildren` with multiple child ids
- `pruneClaimsForChildren` no-match case

## Scope

- Add focused unit tests for `resolveDelegateTarget` edge branches.
- Add session-service integration-style tests for stopped tombstones and prune claim GC.
- Keep the tests local to core; broader CLI/plugin workflow coverage belongs in separate issues.

## Acceptance Criteria

- Behavior-bearing core branches introduced by this work are pinned by tests.
- Stopped and completed terminal child tombstones are both covered.
- Claim pruning behavior is covered for multiple matches and no matches.

### PR DESCRIPTION
**Pull request by tobyhede**
Fixes three entangled delegation/collect issues. Core owns the runbook logic; the CLI consumes it as a thin wrapper.

## #400 — Frame-aware, core-owned delegate frontier
`rd delegate` (bare) reconstructed the delegate frontier inside the CLI (`deriveFrontierFromState`), which is runbook logic and was at risk of leaking a token from another FOR-loop frame.

- Added `deriveDelegateFrontier(state)` to `@rundown-org/core` (`delegation-inference.ts`). It derives the frontier from typed per-substep delegation records, scoped to the **active frame**, so a repeated-frame/FOR case cannot surface a token from another frame.
- Removed the CLI-local reconstruction; `delegate.ts` now calls the core helper. The CLI stays a thin wrapper.
- Coverage moved to core (`delegation-inference.test.ts`) with multiple `substepStates` sharing a substep id across different frames.

Closes #400

## #397 — Narrowed bare `rd collect` idempotency
Bare `rd collect` on a non-DELEGATE cursor reported `already-aggregated` whenever **any** delegation record existed anywhere in the run, masking misuse on ordinary non-DELEGATE steps that were never delegated-aggregation successors.

- Added core signal `isPostDelegateAggregationCursor(state, steps)`: the idempotent no-op is valid only when the cursor's document-order predecessor is an **aggregated DELEGATE step** (its delegate substeps are all `done`).
- `collect.ts` replaces the unscoped `hasPriorDelegation` heuristic with this core-derived signal.
- Bare `rd collect` on an unrelated ordinary step now errors `NOT_DELEGATE_STEP`; idempotency **after** a real delegated aggregation is preserved.
- Tests in `collect.test.ts` cover both the positive (post-auto-aggregation idempotent) and negative (`NOT_DELEGATE_STEP` on an ordinary step further past the aggregated DELEGATE step) paths.

Closes #397

## #402 — Expanded core edge-case coverage (tests only)
No production change required — existing core behavior was already correct.

- `resolveDelegateTarget`: active delegation without a frontier token, issuable-over-already-issued precedence, done-substep skipping, active-frame isolation, RD-814 (DELEGATE without a runbook).
- `deriveDelegateFrontier`: frame isolation, order-independence, non-FOR single frame, cancelled/token-less exclusion, `activeFrameKey` fallback.
- `isPostDelegateAggregationCursor`: positive/negative/first-step/not-yet-aggregated/absent-cursor-step.
- `SessionService`: stopped **and** completed terminal child tombstones, `pruneClaimsForChildren` over multiple child ids, and the no-match no-op.

Closes #402

## Verification
`npm run verify` passes cleanly (format, spell, lint, types, docs, full unit suite). Core +19 tests, CLI net result accounts for the moved frontier suite.

## Summary by CodeRabbit

* **Bug Fixes**
  * Collect command now correctly errors on non-DELEGATE steps instead of reporting a false idempotent success.

* **Improvements**
  * More robust delegation handling so delegation targets and aggregation detection behave consistently across frames and iterations.

* **Tests**
  * Added regression test for collect behavior on non-DELEGATE steps.
  * Expanded delegation frontier and aggregation inference tests.
  * Added session service tests for delegation claim lifecycle and pruning.

### CODE REVIEW

File: packages/core/src/runbook/delegation-inference.ts [hunk 1]
```diff
@@ -226,6 +226,101 @@ export function resolveDelegateTarget(
...
+  if (delegateSubsteps.length === 0) return false;
+
+  // Aggregation evidence: every delegate substep of the predecessor reached
+  // `done` in some recorded frame. SubstepStates accumulate per (id, frameKey),
+  // so a frame-agnostic existence check is sufficient to confirm the predecessor
+  // DELEGATE step actually aggregated (any frame having all substeps done).
+  const substepStates = state.substepStates ?? [];
+  return delegateSubsteps.every((substep) =>
+    substepStates.some((ss) => ss.id === substep.id && ss.status === 'done'),
+  );
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/src/runbook/delegation-inference.ts]: Confirmed and fixed in 115bcb0e. The frame-agnostic `some(ss.id === ... && status === done)` check could indeed return `true` when no single predecessor frame fully aggregated -- e.g. delegate substeps reaching `done` across different FOR-iteration frames, or `done` records from an unrelated step that reuses the same substep ids.

The matching is now frame-scoped: it collects the candidate frames that (a) belong to the predecessor step (via a new `isFrameForStep` helper matching the exact `<step>` segment of the `<step>|<iteration>` frame key, so step `1` does not collide with step `12`) and (b) contain a `done` delegate substep, then requires that *some single* such frame has *all* delegate substeps `done`. A mix across frames, or evidence from a different step, no longer qualifies, so a bare `rd collect` there correctly yields `NOT_DELEGATE_STEP`.

Added 3 core tests in `delegation-inference.test.ts`: false when delegate substeps are `done` only across different frames; false when `done` records belong to a different step reusing substep ids; and true when a single FOR-iteration frame fully aggregated (idempotency preserved, including a lingering noise record in another frame). The existing positive idempotency cases still pass. `npm run verify` is green.

File: packages/core/src/runbook/delegation-inference.ts [hunk 2]
```diff
@@ -226,6 +226,139 @@ export function resolveDelegateTarget(
...
+  }
+
+  for (const frameKey of candidateFrames) {
+    const fullyAggregated = delegateSubsteps.every((substep) =>
+      substepStates.some(
+        (ss) => ss.id === substep.id && ss.frameKey === frameKey && ss.status === 'done',
+      ),
+    );
+    if (fullyAggregated) return true;
+  }
```
[the remark this answers is not in the document]
Comment (tobyhede) [reply packages/core/src/runbook/delegation-inference.ts:357]: @coderabbitai I looked into this and am declining the change — the targeted false-positive is not reachable, and the proposed fix would regress intended, tested behavior.

**1. The proposed `cursorFrameKey` filter regresses a documented path.** The unit test at `delegation-inference.test.ts:587` ("returns true when a single FOR-iteration frame fully aggregated") covers a DELEGATE step inside a FOR loop that aggregates in frame `1|2`, after which the cursor advances to plain step `2`. That successor carries **no iteration segment**, so `state.activeFrameKey ?? buildFrameKey(state.step)` resolves to `2|` (empty iteration). Filtering `candidateFrames` to that iteration would drop the predecessor's `1|2` frame and flip the result to `false`, breaking the legitimate "delegate-in-loop → advance to following step" idempotency case.

**2. The stale-iteration false-positive is not reachable.** The check is position-based on document order. For the cursor to sit on the successor *in iteration N*, the machine must already have run and aggregated the predecessor DELEGATE *in iteration N* — so frame `N` is itself fully aggregated and the `true` is correct. A lingering frame `M ≠ N` in `substepStates` never produces a *wrong* `true`, because the cursor's own iteration frame is already complete. The genuine hazard (delegate substeps `done` split across different frames with no single frame complete) is already guarded — see `delegation-inference.test.ts:571`.

**3. Minimal-change / unrepresentable-state.** Defending a state the state machine does not produce, at the cost of breaking a real tested path, is net-negative here.

Caveat I considered: a `goto <successor> --index N` jump could in principle land the cursor on a later-iteration successor without that iteration aggregating. But the docstring already states GOTO-reached steps are not idempotent successors, and a position-based heuristic fundamentally cannot distinguish jump-arrival from advance-arrival regardless of frame filtering — so the proposed filter would not resolve that cleanly while *also* regressing case #1. That's a broader design question, not something this minimal patch fixes.

Verdict: skipping. No production change; `isPostDelegateAggregationCursor` suite stays green (10/10).
